lib/src/widget/crop_result.dart

```diff
@@ -0,0 +1,18 @@
+// Created by alex@justprodev.com on 04.03.2024.
+
+import 'dart:typed_data';
+
+sealed class CropResult {
+  const CropResult();
+}
+
+class CropSuccess extends CropResult {
+  const CropSuccess(this.croppedImage);
+  final Uint8List croppedImage;
+}
+
+class CropFailure extends CropResult {
+  const CropFailure(this.error, [this.stackTrace]);
+  final Object error;
+  final StackTrace? stackTrace;
```

Comment:
I found my past mistake that errors happened in crop_your_image should not be `Error` but `Exception`. Because `Error` is for mistakes in programming while `Exception` is for some unexpected situation at runtime. 

Considering that, this part should be like below.

```suggestion
  final Exception exception;
```

But this requires additional fixes, so I'll accept and merge this PR and address the issue above in another PR by myself. Thank you for making me find it.

Reply:
Yeah, exception is standard for languages like Java, because only exception can be used in `try/catch`. But dart can throw anything and object with trace is standard for dart at the current moment. We can't change this and should use it ;).

https://dart.dev/language/error-handling

Reply:
I think `object+stacktrace` should be used at low-level (in libraries).

Abstractions like exceptions are better for programs.

And yes, Error is another standard for fatal situations where we can't continue to processing.

Anyway, CropFailure is wrapper of any cause - Exception or Error. May be `error` should be renamed to `cause`.

Reply:
Thanks. I'll check the link and consider what should be exposed from the package and what should be handled internally. The idea of renaming `error` to `cause` sounds nice!